tests/src/lease/safe_delivery.rs

```diff
@@ -0,0 +1,294 @@
+use std::{cell::Cell, slice};
+
+use anyhow::anyhow;
+use currency::Currency as _;
+use finance::coin::Amount;
+use lease::api::ExecuteMsg;
+use osmosis_std::types::osmosis::gamm::v1beta1::{
+    MsgSwapExactAmountIn, MsgSwapExactAmountInResponse,
+};
+use platform::trx;
+use sdk::{
+    cosmos_sdk_proto::{ibc::applications::transfer::v1::MsgTransfer, traits::TypeUrl as _},
+    cosmwasm_std::{Addr, Binary, Coin as CwCoin, WasmMsg},
+    cw_multi_test::AppResponse,
+    neutron_sdk::sudo::msg::{RequestPacket, SudoMsg as NeutronSudoMsg},
+};
+
+use crate::common::{
+    cwcoin,
+    test_case::{
+        app::{Action, ConfigurableWasmBuilder, Request, Wasm as WasmTrait},
+        response::{RemoteChain, ResponseWithInterChainMsgs},
+        TestCase,
+    },
+    ADMIN, USER,
+};
+
+use super::{create_test_case_with_wasm, open_lease, Lpn, PaymentCoin, PaymentCurrency};
+
+#[test]
+fn safe_delivery() {
+    const ERROR_MESSAGE: &str = "Simulated error";
+
+    let forward_specific_execute: Cell<bool> = Cell::new(true);
+    let lease_addr_cell: Cell<&str> = Cell::new("DEADCODE");
```

Comment:
please refrain from using Cell and RefCell types if their usage is really needed!

Reply:
For the boolean, I could've used atomic or a channel, but I felt like that would be an overkill for this use.
For the lease address, same thing applies but most likely will have to include unsafe Rust if channel or a `Mutex` is not used.

Reply:
Also, that's the tests and not production code. I just feel like it would have been a bit easier to not complicate it and keep it simple.

Reply:
exactly the same as what we need ... and having interior mutable data and reference counting data is a step in the opposite direction. Could we `CwApp::store` each time we want to change the behavior?

Reply:
`CwApp::store` is specialized to only work with their own WasmKeeper. Thus the need for this PR.